### PR TITLE
Define mechanism for Camel Connector/Assembly deployment to K8s

### PR DESCRIPTION
**Pull request by fhcarter**
Fixes #454

As part of creating the assembly, define a service procedure that can deploy the associated Camel Connector to a K8sCluster.  Each assembly now includes such a service which is largely implemented by a single such service procedure defined in the Camel Connector.  The assembly versions collect parameter values and provide their generated source name.

Documentation added to associated READMEs for both.  Once merged, will push connector image to quay.io so that this will all work

### CODE REVIEW

**Comment by fhcarter** [comment]
[Jenkins Run](https://jenkins.vantiq.com/jenkins/job/vantiq-extension-sources-branch/206/)

**Comment by fhcarter** [comment]
Note: We do this via a procedure so that users don't have to copy values from documentation into the UI tool.  I'll probably add a request to have the ability to fill that tool's form from a Vantiq document & add the ability generate such a document in the assembly) which may make things a bit easier.   But that can come later.  Rather get things out & available first.

File: build.gradle [hunk 1]
```diff
@@ -125,6 +125,19 @@ subprojects {
     if (project.tasks.findByName('buildConnectorImage')) {
         buildConnectorImage.dependsOn assemble
     }
+    project.afterEvaluate { x ->
+        // Add dependency to Gradle for the archive file used in the connector docker image. This avoids warnings
+        // from gradle about undeclared dependencies.  The warning didn't really cause an issue, but it looks messy
+        // and is better dealt with properly.
+
+        // We have to check this later in the gradle process since the copyConnectorBuildFiles is generated later in
+        // the process (by the dcompose plugin). Slso, some projects may not have a distTar task, so only generate
```
Comment (jcguion) [inline build.gradle:134]:
```suggestion
        // the process (by the dcompose plugin). Also, some projects may not have a distTar task, so only generate
```

File: buildSrc/src/main/groovy/io/vantiq/extsrc/assy/tasks/AssemblyResourceGeneration.groovy [hunk 2]
```diff
@@ -440,6 +471,96 @@ class AssemblyResourceGeneration extends DefaultTask {
...
+     *
+     * This incorporates the deployment procedure created by addDeploymentProcedure().
+     *
+     * @param kameletAssemblyDir Path location of the base location for the project artifacts
+     * @param packageName String name of the package to which this service will belong
+     * @param kamName String name of the kamelet which will be used to construct the service name
+     * @param serviceName String name of the service we're creating
+     * @param interfaceSpec List<Map> service's interface definition
+     * @returns Map<String, Object> containing Path
+     * written &
```
Comment (jcguion) [inline buildSrc/src/main/groovy/io/vantiq/extsrc/assy/tasks/AssemblyResourceGeneration.groovy:530]: nit - LF formatting

File: buildSrc/src/main/groovy/io/vantiq/extsrc/assy/tasks/AssemblyResourceGeneration.groovy [hunk 3]
```diff
@@ -577,8 +684,23 @@ class AssemblyResourceGeneration extends DefaultTask {
             type: 'dev',
             v: 5,
             isModeloProject: true,
-            exclusionList: [ "type/${CAMEL_MESSAGE_SCHEMA}",
-                             "sourceimpl/${CAMEL_SOURCE_TYPE}",
+
+            // Add exclusions for thing we bring in from other assemblies.  We don't wan to include them in our
```
Comment (jcguion) [inline buildSrc/src/main/groovy/io/vantiq/extsrc/assy/tasks/AssemblyResourceGeneration.groovy:688]:
```suggestion
            // Add exclusions for thing we bring in from other assemblies.  We don't want to include them in our
```

File: buildSrc/src/main/groovy/io/vantiq/extsrc/assy/tasks/AssemblyResourceGeneration.groovy [hunk 4]
```diff
@@ -577,8 +684,23 @@ class AssemblyResourceGeneration extends DefaultTask {
...
+            // definition -- we want there to be only one copy/instance.
```
Comment (jcguion) [inline buildSrc/src/main/groovy/io/vantiq/extsrc/assy/tasks/AssemblyResourceGeneration.groovy:689]:
```suggestion
            // definition -- we want them to be only one copy/instance.
```

File: buildSrc/src/main/groovy/io/vantiq/extsrc/assy/tasks/AssemblyResourceGeneration.groovy [hunk 5]
```diff
@@ -1018,15 +1173,32 @@ class AssemblyResourceGeneration extends DefaultTask {
...
-        return '/' + VANTIQ_SYSTEM_PREFIX + "${resourceType}/${resourceId}"
+        if (GRANDFATHERED_TYPES.contains(resourceType)) {
+            return '/' + "${resourceType}/${resourceId}"
+        } else {
+            return '/' + VANTIQ_SYSTEM_PREFIX + "${resourceType}/${resourceId}"
+        }
+    }
+
+    /**
+     * Build a resourceReference-like think for use in exclusion lists.
```
Comment (jcguion) [inline buildSrc/src/main/groovy/io/vantiq/extsrc/assy/tasks/AssemblyResourceGeneration.groovy:1192]:
```suggestion
     * Build a resourceReference-like thing for use in exclusion lists.
```

File: camelConnector/README.md [hunk 6]
```diff
@@ -92,25 +96,26 @@ If you wish to change this, you will need to perform the following actions:
...
+
 To change the logging settings, edit the logging config file
 `<install location>/camelConnector/src/main/resources/log4j2.xml`,
 which is an [Apache Log4j configuration file](https://logging.apache.org/log4j/2.x/manual/configuration.html). The logger 
 name for each class is the class's fully qualified class name, *e.g.* "io.vantiq.extjsdk.ExtensionWebSocketClient".  
 
 <a name="serverConfig" id="serverConfig"></a>
-## Server Config File
-(Please read the [SDK's server config documentation](../extjsdk/README.md#serverConfig) first.)
+###lease read the [SDK's server config documentation](../extjsdk/README.md#serverConfig) first.)
```
Comment (jcguion) [inline camelConnector/README.md:107]: `Server Config File` truncated

File: camelConnector/README.md [hunk 7]
```diff
@@ -300,6 +305,94 @@ Some tests may require other gradle properties as well.
...
+
+* **clusterName** -- this is the name of the Vantiq K8sCluster to which to deploy the connector.
+* **installation** -- this is the name of the installation to be deployed.
+* **sourceName** -- the name of the Vantiq Source that this connector will implement
+
+Optional Values:
+
+* **k8sNamespace** -- the namespace within the Kubernetes cluster into which to place the installation.
+* **targetUrl**, **accessToken**, **cpuLimit**, **memoryLimit**, **connectorImageTag** -- described 
+  as part of the [assembly installatio process](#assemblyInstallation).  _Note that there are system provided or
```
Comment (jcguion) [inline camelConnector/README.md:389]:
```suggestion
  as part of the [assembly installation process](#assemblyInstallation).  _Note that there are system provided or 
```

File: camelConnector/README.md [hunk 8]
```diff
@@ -300,6 +305,94 @@ Some tests may require other gradle properties as well.
...
+* **installation** -- this is the name of the installation to be deployed.
+* **sourceName** -- the name of the Vantiq Source that this connector will implement
+
+Optional Values:
+
+* **k8sNamespace** -- the namespace within the Kubernetes cluster into which to place the installation.
+* **targetUrl**, **accessToken**, **cpuLimit**, **memoryLimit**, **connectorImageTag** -- described 
+  as part of the [assembly installatio process](#assemblyInstallation).  _Note that there are system provided or 
+  determined values for most of these, the **accessToken** (or **vantiqAccessToken**) must have a value provided 
+  either an invocation or installation time._
```
Comment (jcguion) [inline camelConnector/README.md:391]:
```suggestion
  either at invocation or installation time._
```

**Review by jcguion — APPROVED** [review]
LGTM

**Pull request merged**